app/backends/workflow-orchestration.ts

```diff
@@ -81,23 +85,99 @@ async function _handleWorkItemResults(
 }
 
 /**
- * Creates a work item that uses all the output of the previous step.
+ * Read a STAC catalog and return the item links. This does not handle sub-catalogs. This function 
+ * makes assumptions based on the Harmony STAC directory layout for services inputs/outputs and
+ * is only intended to be used when aggregating service outputs into a single catalog.
+ * @param catalogPath - the path to the catalog
+ */
+async function getLinksFromCatalog(catalogPath: string): Promise<StacItemLink[]> {
+  const baseDir = path.dirname(catalogPath).replace(env.hostVolumePath, PATH_TO_CONTAINER_ARTIFACTS);
+  const text = (await fs.readFile(catalogPath)).toString();
+  const catalog = JSON.parse(text);
+  const links: StacItemLink[] = [];
+  for (const link of catalog.links) {
+    if (link.rel === 'item') {
+      // make relative path absolute
+      const { href } = link;
+      link.href = `${baseDir}/${path.normalize(href)}`;
+      links.push(link);
+    }
+  }
+
+  return links;
+}
+
+/**
+ * Creates a work item that uses all the output of the previous step. This function assumes that
+ * all work items for the previous step are completed. It also relies on the convention that
+ * services write out their results as STAC catalogs with the following path
+ * `/tmp/<JOB_ID>/<WORK_ITEM_ID>/outputs /catalogN.json`
+ * where N is from 0 to the number of results - 1.
  * 
  * @param tx - The database transaction
  * @param currentWorkItem - The current work item
  * @param nextStep - the next step in the workflow
- * @param results - an array of paths to STAC catalogs
+ * @param results - an array of paths to STAC catalogs from the last worked item
  */
 async function createAggregatingWorkItem(
-  tx: Transaction, currentWorkItem: WorkItem, nextStep: WorkflowStep, results: string[],
+  tx: Transaction, currentWorkItem: WorkItem, nextStep: WorkflowStep,
 ): Promise<void> {
-  // TODO use ALL outputs from the prior step here (HARMONY-961)
-  // just use the first result of the current output for now
+  const itemLinks: StacItemLink[] = [];
+  // get all the previous results
+  const workItemCount = await workItemCountForStep(tx, currentWorkItem.jobID, nextStep.stepIndex - 1);
+  let page = 1;
+  let processedItemCount = 0;
+  while (processedItemCount < workItemCount) {
+    const prevStepWorkItems = await getWorkItemsByJobIdAndStepIndex(tx, currentWorkItem.jobID, nextStep.stepIndex - 1, page);
+    // guard against failure case where we cannot retrieve all items - THIS SHOULD NEVER HAPPEN
+    if (prevStepWorkItems.workItems.length < 1) break;
```

Comment:
Throw an exception instead of break? I think at a minimum we should log something if this were to occur.

Reply:
I break because I added a check below that will throw an exception if the number of processed items is not what we expect. Better to check there to catch any situation where we did not get all the worked items.

Reply:
I think this is more likely in the future.  We have cases in the pipeline where a variable subset is requested but some data files don't contain the variable, which may result in this case being triggered, and a success-with-warning condition.